### PR TITLE
JBPM-8374 Added support for invoking Decision Services from BPMN Business Rule Tasks

### PR DESCRIPTION
**Pull request by SimonRinguette**
I've update the business rule task code to allow invocation of DMN decision services.

I'm also contributing a test case that tests that only the output decisions of the decision service are returned and that the encapsulated ones are not. The test case also show that the rest of the model is not executed as the rest of the model has an input that is not provided.

### CODE REVIEW

**Comment by kie-ci5** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

**Comment by SimonRinguette** [comment]
FYI @tarilabs @etirelli

**Comment by etirelli** [comment]
@mswiderski @romartin 

FYI, in DMN one can invoke a "decision" or a "decision service". 

Invoking a decision is already supported by the business rule task implementation.

A "decision service" is a specific node of a DMN model that defines a decision service and its parameters. This PR from @SimonRinguette introduces support for invoking such decision services.

Can you please review and merge?

Thanks.

**Comment by tarilabs** [comment]
@mswiderski I have created https://issues.jboss.org/browse/JBPM-8374 as requested.

I don't have the github admin rights to change this PR title (which in turn reflects in the commit message) but I believe you can do it directly on your side while merging, please?

Thanks; available if any further actions required

**Comment by SimonRinguette** [comment]
@mswiderski @tarilabs I've update the PR name and the commit message

**Comment by romartin** [comment]
I think this has no impact on the process designer side, so looks good to me... anyway can you please @Josephblt confirm the process editor side (AFAIK you added those properties)? 
Also better asking for a quick double check to @manstis @jomarko 
Thanks!

File: jbpm-workitems/jbpm-workitems-bpmn2/src/test/java/org/jbpm/process/workitem/bpmn2/BusinessRuleTaskTest.java [hunk 1]
```diff
@@ -167,6 +167,34 @@ public void testDecisionPassingNonNullToDMN() throws Exception {
...
 
+    @Test
+    public void testCallingDecisionService() throws Exception {
+        //This test make sure that null variables are passed to the DMN execution context, otherwise DMN will throw an exception that an input requirement is missing
+        KieBase kbase = readKnowledgeBase();
+        KieSession ksession = createSession(kbase);
+
+        BusinessRuleTaskHandler handler = new BusinessRuleTaskHandler(GROUP_ID,
+                                                                      ARTIFACT_ID,
+                                                                      VERSION);
```
Comment (jomarko) [inline jbpm-workitems/jbpm-workitems-bpmn2/src/test/java/org/jbpm/process/workitem/bpmn2/BusinessRuleTaskTest.java]: This `handler` is not used at all, please remove it.

**Comment by Josephblt** [comment]
Hello @romartin. This will not cause any problems in the Process Editor. 

Hello @jomarko, @manstis. The user actually can add a Decision Service by manually adding it as a Data Input. As a matter of fact all this properties: Namespace, Decision Name and DMN Model Name, are data inputs that are hidden by code. We should consider also having a UI property for Decision Service, or go back to having them all as Data Inputs (the way the old designer, did).

But anyway, this discussion is unrelated to this PR, so it looks good to me as well.

File: jbpm-workitems/jbpm-workitems-bpmn2/src/test/java/org/jbpm/process/workitem/bpmn2/BusinessRuleTaskTest.java (hunk 1 above)
Comment (SimonRinguette) [reply jbpm-workitems/jbpm-workitems-bpmn2/src/test/java/org/jbpm/process/workitem/bpmn2/BusinessRuleTaskTest.java]: I've added a commit to correct this. Thanks

**Comment by mswiderski** [comment]
so everyone is happy and we can merge this or?

**Comment by mswiderski** [comment]
Jenkins retest this

**Pull request merged**